subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -12,7 +12,9 @@ LOG_MODULE_REGISTER(net_wifi_mgmt, CONFIG_NET_L2_NRF_WIFI_MGMT_LOG_LEVEL);
 #include <zephyr/net/net_core.h>
 #include <zephyr/net/net_if.h>
 #include <zephyr/net/wifi_mgmt.h>
+#ifdef CONFIG_WPA_SUPP
```

Comment:
Why should it be included conditionally?

Reply:
> Why should it be included conditionally?
As WPA supplicant is disabled, the supplicant API header file
should not be included.

Reply:
This is not an explanation and including the header should not be an issue.

Reply:
As @D-Triveni  was saying above, It will not work, as the header file `supp_api.h` is not included if `CONFIG_WPA_SUPP=n`.